homeassistant/components/remootio/cover.py

```diff
@@ -0,0 +1,191 @@
+        """Initialize this cover entity."""
+        super().__init__()
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_device_class = device_class
+        self._remootio_client = remootio_client
+        self._attr_device_info = DeviceInfo(
+            name="Remootio",
+            manufacturer="Assemblabs Ltd",
+            suggested_area="The first end-to-end encrypted Wi-Fi and Bluetooth enabled smart remote controller, "
+            "that lets you control and monitor your gates and garage doors using your smartphone.",
+        )
+        self._attr_should_poll = False
+        self._attr_supported_features = cover.SUPPORT_OPEN | cover.SUPPORT_CLOSE
+
+    async def async_added_to_hass(self) -> None:
+        """Register listeners to the used Remootio client to be notified on state changes and events."""
+        await self._remootio_client.add_state_change_listener(
+            RemootioCoverStateChangeListener(self)
+        )
+
+        await self._remootio_client.add_event_listener(RemootioCoverEventListener(self))
+
+        self.async_schedule_update_ha_state(force_refresh=True)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Terminate the used Remootio client."""
+        await self._remootio_client.terminate()
+
+    async def async_update(self) -> None:
+        """Trigger state update of the used Remootio client."""
+        await self._remootio_client.trigger_state_update()
+
+    @property
+    def is_opening(self):
+        """Return True when the Remootio controlled garage door or gate is currently opening."""
+        return self._remootio_client.state == State.OPENING
+
+    @property
+    def is_closing(self):
+        """Return True when the Remootio controlled garage door or gate is currently closing."""
+        return self._remootio_client.state == State.CLOSING
+
+    @property
+    def is_closed(self):
+        """Return True when the Remootio controlled garage door or gate is currently closed."""
+        return self._remootio_client.state == State.CLOSED
+
+    def open_cover(self, **kwargs: Any) -> None:
+        """Open the Remootio controlled garage door or gate."""
+        self.hass.async_create_task(self.async_open_cover())
+
+    async def async_open_cover(self, **kwargs):
+        """Open the Remootio controlled garage door or gate."""
+        await self._remootio_client.trigger_open()
+
+    def close_cover(self, **kwargs: Any) -> None:
+        """Close the Remootio controlled garage door or gate."""
+        self.hass.async_create_task(self.async_close_cover())
+
+    async def async_close_cover(self, **kwargs):
+        """Close the Remootio controlled garage door or gate."""
+        await self._remootio_client.trigger_close()
+
+
+class RemootioCoverStateChangeListener(Listener[StateChange]):
+    """Listener to be invoked when Remootio controlled garage door or gate changes its state."""
+
+    __owner: RemootioCover
```

Comment:
Seems you missed this double underscore.

Reply:
Adapted.